src/components/input-fields/simple-input/SimpleInput.stories.jsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { SimpleInput } from "./SimpleInput";
+import { WithLabel } from "../with-label/WithLabel";
+import { WithErrorMessage } from "../with-error-message/WithErrorMessage";
+import { WithTranslationBadge } from "../with-translation-badge/WithTranslationBadge";
+import FieldContext from "../field-context/FieldContextState";
+import { useForm, FormFieldWrapper, Form } from "../../form/index";
+import GermanFlagIcon from "./GermanFlagIcon";
+
+function SimpleInputDemo() {
+  const { control, handleSubmit, getFieldErrors, validate } = useForm();
+
+  const [formData, setFormData] = React.useState(null);
+
+  const onSubmit = async (data) => {
+    if (await validate()) {
+      console.log(data);
+      setFormData(data);
+    }
+  };
+
+  const getErrorList = (fieldName) => {
+    return getFieldErrors(fieldName)?.map(({ message }) => message) || [];
+  };
+
+  return (
+    <>
+      <Form style={{ display: "flex", flexDirection: "column", gap: 8 }} onSubmit={handleSubmit(onSubmit)}>
+        <div style={{ display: "flex", gap: 8 }}>
+          <FormFieldWrapper
+            rules={[
+              { name: "required", condition: (value) => value && value !== "", message: "This field is required" },
+            ]}
+            control={control}
+            name={`formData.name`}
+          >
+            <FieldContext>
+              <WithLabel active={true} content="Name*" />
+              <WithErrorMessage active={true} content={getErrorList("formData.name")} />
+              <SimpleInput state={getErrorList("formData.name").length > 0 ? "error" : "success"} />
+            </FieldContext>
+          </FormFieldWrapper>
+
+          <FormFieldWrapper control={control} name={`formData.value`}>
+            <FieldContext>
+              <WithTranslationBadge active={true} content={{ flag: GermanFlagIcon, country: "DE" }} />
+              <SimpleInput />
+              <WithLabel active={true} content="Value" />
+            </FieldContext>
+          </FormFieldWrapper>
+        </div>
+
+        <div style={{ display: "flex", gap: 8 }}>
+          <FormFieldWrapper
+            rules={[
+              { name: "required", condition: (value) => value && value !== "", message: "This field is required" },
+            ]}
+            control={control}
+            name={`formData.name1`}
+          >
+            <FieldContext>
```

Comment:
Looks like this is intended to be a HOC but is modifying the state to add stuff to the component that should be par of the definition of the component. I don't think adding enhancers from the state is a good pattern.